### PR TITLE
implement unfollow handler

### LINKED ISSUE #19: Create unfollow handler
## Summary

From the spec:
```
unfollow:
  input: -
  action: remove "follow" event
  return: -
```

## Requirements

- [x] Update table to allow 'remove' operation
- [x] Test table 'remove' operation
- [x] Create handler
- [x] Test handler
- [ ] After-review additional treatment: send an `unfollow` message to other followers (not self of course).

### PR DESCRIPTION
**Pull request by ThomasJuster**
Fixes #19

### CODE REVIEW

File: src/threads/unfollow.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  try {
+    const [ followEvent ] = await forumTable.getFollowers({
+      participantId,
+      itemId,
+      limit: 1,
+      filters: { userId, connectionId },
+    });
+    if (!followEvent) return notFound();
+
+    await forumTable.removeThreadEvent(followEvent);
```
Comment (smadbe) [inline src/threads/unfollow.ts:25]: There is no need to check anything. Just request the "follow" event removal. 
This should work (for your tests) either if the record exists or not, you'll never know if it exists, but that's not a problem. (anyway this is called when leaving the thread, the frontend does not care about the response)

File: src/threads/unfollow.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    return ok();
```
Comment (smadbe) [inline src/threads/unfollow.ts:27]: That's strange to broadcast the "follow" without broadcasting the "unfollow". I know, the reason is that "unfollow" is not an event for now... Maybe it should be sent anyway... but that means message type does not correspond to event type 😕... do whatever sound best to you for now, but just keep that in mind.

**Review by smadbe — APPROVED** [review]

File: src/threads/unfollow.ts (hunk 1 above)
Comment (ThomasJuster) [reply src/threads/unfollow.ts:25]: I need to know the sort key of the event to remove, that's why I search for it first.
I'll remove the not-found early return and update the tests accordingly

File: src/threads/unfollow.ts (hunk 2 above)
Comment (ThomasJuster) [reply src/threads/unfollow.ts:27]: Messages can be something else/more than thread events, I'll add another kind of message `UnfollowEvent` and send it in the unfollow handler. And I'll update the test as well.

File: src/threads/unfollow.ts (hunk 1 above)
Comment (ThomasJuster) [reply src/threads/unfollow.ts]: Fixed

File: src/threads/unfollow.ts (hunk 2 above)
Comment (ThomasJuster) [reply src/threads/unfollow.ts:34]: Fixed

File: src/threads/unfollow.ts (hunk 1 above)
Comment (smadbe) [reply src/threads/unfollow.ts]: I don't think you need the sort key to remove it, primary key and criterions should be enough. Doing two requests for such an operation is a bit against the way of working with such dbs.

**Pull request merged**